File: frontend/lib/models/vendor.dart
// lib/models/vendor.dart

class Vendor {
  final int? id;
  final String name;
  final String phone;
  final String? email;
  final String? gst;
  final String? address;

  Vendor({
    this.id,
    required this.name,
    required this.phone,
    this.email,
    this.gst,
    this.address,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) {
    return Vendor(
      id: json['id'],
      name: json['name'],
      phone: json['phone'],
      email: json['email'],
      gst: json['gst'],
      address: json['address'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
      'email': email,
      'gst': gst,
      'address': address,
    };
  }
}

File: backend/models/vendor.py
# heymachi_backend/models/vendor.py

from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from .base import Base

class Vendor(Base):
    __tablename__ = "vendors"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    phone = Column(String, nullable=False, unique=True)
    email = Column(String, nullable=True)
    gst = Column(String, nullable=True)
    address = Column(String, nullable=True)
    business_id = Column(String, ForeignKey("business_accounts.id"))
    
    business = relationship("BusinessAccount", backref="vendors")

